modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -209,23 +228,27 @@ export class MessengerService {
       return next()
     }
 
-    const messageType = event.type === 'default' ? 'text' : event.type
-    const messenger = this.getMessengerClientByBotId(event.botId)
+    const context: MessengerContext = {
+      bp: this.bp,
+      event,
+      client: this.getMessengerClientByBotId(event.botId),
+      handlers: [],
+      payload: _.cloneDeep(event.payload),
+      botUrl: process.EXTERNAL_URL,
+      messages: []
+    }
 
-    if (!_.includes(outgoingTypes, messageType)) {
-      return next(new Error(`Unsupported event type: ${event.type}`))
+    for (const renderer of this.renderers) {
+      if (renderer.handles(context)) {
+        renderer.render(context)
+        context.handlers.push(renderer.id)
+      }
     }
 
-    if (messageType === 'typing') {
-      const typing = parseTyping(event.payload.value)
-      await messenger.sendAction(event.target, 'typing_on')
-      await Promise.delay(typing)
-      await messenger.sendAction(event.target, 'typing_off')
```

Comment:
If you take a look at the `sendAction` function below, you can see that there are some configuration check in order to comply with Facebook Messenger restrictions regarding GDPR in Europe:

```ts
    ...
    const config = await this.getConfig()
    if (config.disabledActions?.includes(action)) {
      debugMessages('outgoing action skipped (blacklisted)', { action })
      return
    }
    ...
```

This validation is missing in your typing sender. Can you simply append this code to it?